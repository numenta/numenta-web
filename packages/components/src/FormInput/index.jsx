// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const FormInput = ({name, onChange, placeholder, stretch, type, value}) => {
  const classes = [styles.formInput]

  if (stretch) classes.push(styles[stretch])

  return (
    <input
      className={classNames(...classes)}
      id={name}
      name={name}
      onChange={onChange}
      placeholder={placeholder}
      type={type}
      value={value}
    />
  )
}

FormInput.propTypes = {
  name: React.PropTypes.string.isRequired,
  onChange: React.PropTypes.func,
  placeholder: React.PropTypes.string,
  stretch: React.PropTypes.oneOf(['small', 'medium', 'large']),
  type: React.PropTypes.string.isRequired,
  value: React.PropTypes.string,
}

FormInput.defaultProps = {
  stretch: 'large',
}

export default FormInput
