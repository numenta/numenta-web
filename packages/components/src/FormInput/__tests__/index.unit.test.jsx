import React from 'react'
import renderer from 'react-test-renderer'

import FormInput from '../../FormInput'


describe('FormInput React component', () => {

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <FormInput
        name="field"
        onChange={() => {}}
        placeholder="Default"
        type="text"
        value="Contents"
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Small stretch correctly', () => {
    const component = renderer.create(
      <FormInput
        name="field"
        onChange={() => {}}
        placeholder="Default"
        stretch="small"
        type="text"
        value="Contents"
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Medium stretch correctly', () => {
    const component = renderer.create(
      <FormInput
        name="field"
        onChange={() => {}}
        placeholder="Default"
        stretch="medium"
        type="text"
        value="Contents"
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Large stretch correctly', () => {
    const component = renderer.create(
      <FormInput
        name="field"
        onChange={() => {}}
        placeholder="Default"
        stretch="large"
        type="text"
        value="Contents"
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
