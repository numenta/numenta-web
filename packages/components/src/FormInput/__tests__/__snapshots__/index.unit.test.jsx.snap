exports[`FormInput React component Renders Default correctly 1`] = `
<input
  className="formInput large"
  id="field"
  name="field"
  onChange={[Function]}
  placeholder="Default"
  type="text"
  value="Contents" />
`;

exports[`FormInput React component Renders Large stretch correctly 1`] = `
<input
  className="formInput large"
  id="field"
  name="field"
  onChange={[Function]}
  placeholder="Default"
  type="text"
  value="Contents" />
`;

exports[`FormInput React component Renders Medium stretch correctly 1`] = `
<input
  className="formInput medium"
  id="field"
  name="field"
  onChange={[Function]}
  placeholder="Default"
  type="text"
  value="Contents" />
`;

exports[`FormInput React component Renders Small stretch correctly 1`] = `
<input
  className="formInput small"
  id="field"
  name="field"
  onChange={[Function]}
  placeholder="Default"
  type="text"
  value="Contents" />
`;
